contrib_bots/lib/tictactoe-bot.py

```diff
@@ -0,0 +1,329 @@
+        else:
+            location = random.choice(blank_locations)
+            row = location[0]
+            col = location[1]
+            board[row][col] = 'o'
+            return board
+
+    def check_validity(self, move):
+        ''' Checks the validity of the coordinate input passed in to make sure it's not out-of-bounds (ex. 5, 5) '''
+        try:
+            split_move = move.split(",")
+            row = split_move[0].strip()
+            col = split_move[1].strip()
+            valid = False
+            if row == "1" or row == "2" or row == "3":
+                if col == "1" or col == "2" or col == "3":
+                    valid = True
+        except IndexError:
+            valid = False
+        return valid
+
+    def sanitize_move(self, move):
+        ''' As there are various ways to input a coordinate (with/without parentheses, with/without spaces, etc.) the
+        input is stripped to just the numbers before being used in the program. '''
+        move = move.replace("(", "")
+        move = move.replace(")", "")
+        move = move.strip()
+        return move
+
+    def tictactoe(self, board, input_string):
+        return_string = ""
+        move = self.sanitize_move(input_string)
+
+        # Subtraction must be done to convert to the right indices, since computers start numbering at 0.
+        row = (int(move[0])) - 1
+        column = (int(move[-1])) - 1
+
+        if board[row][column] != "_":
+            return_string += output_mode("That space is already filled, sorry!", mode)
+            return return_string
+        else:
+            board[row][column] = "x"
+
+        return_string += self.display_board(board)
+
+        # Check to see if the user won/drew after they made their move. If not, it's the computer's turn.
+        if self.win_conditions(board, self.triplets) == True:
+            return_string += output_mode("Game over! You've won!", mode)
+            return return_string
+
+        if self.board_is_full(board) == True:
+            return_string += output_mode("It's a draw! Neither of us was able to win.", mode)
+            return return_string
+
+        return_string += output_mode("My turn:\n", mode)
+        self.computer_move(board)
+        return_string += self.display_board(board)
+
+        # Checks to see if the computer won after it makes its move. (The computer can't draw, so there's no point
+        # in checking.) If the computer didn't win, the user gets another turn.
+        if self.win_conditions(board, self.triplets) == True:
+            return_string += output_mode("Game over! I've won!", mode)
+            return return_string
+
+        return_string += output_mode("Your turn! Enter a coordinate or type help.", mode)
+        return return_string
+
+# -------------------------------------
+flatInitial = sum(initial_board, [])
+def first_time(board):
+    flat = sum(board, [])
+    return True if flat == flatInitial else False
+
+class tictactoeHandler(object):
```

Comment:
Conventionally class names should be in camel case:

https://www.python.org/dev/peps/pep-0008/#class-names